eZ/Publish/API/Repository/Values/Content/Query/Aggregation/Range.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\API\Repository\Values\Content\Query\Aggregation;
+
+use DateTimeInterface;
+use eZ\Publish\API\Repository\Values\ValueObject;
+
+final class Range extends ValueObject
+{
+    /**
+     * Beginning of the range (included).
+     *
+     * @var int|float|\DateTimeInterface|null
+     */
+    private $from;
+
+    /**
+     * End of the range (excluded).
+     *
+     * @var int|float|\DateTimeInterface|null
+     */
+    private $to;
+
+    public function __construct($from, $to)
```

Comment:
This is perfect use case for generics types but for now maybe we can introduce named constructors e.g. `Range::ofInt`, `Range::ofFloat`, `Range::ofDateTime` to guarantee some more type safty?

Reply:
Fixed inhttps://github.com/ezsystems/ezplatform-kernel/pull/94/commits/d9ea47506faf2e21932bd09fa84ca25cccd547e0

Reply:
> This is perfect use case for generics types but for now maybe we can introduce named constructors e.g. `Range::ofInt`, `Range::ofFloat`, `Range::ofDateTime` to guarantee some more type safty?

TBH in this case you should have specific subclasses of Range for different type of data.

Reply:
This will work as well. Will be refactored as follow-up.